Final_Project/FrameHandler.cpp

```diff
@@ -122,6 +123,15 @@ void readCtrlFrame(const char* frame) {
 	int ctrlChar = (int)frame[1];
 	int dcChar = (int)frame[2];
 	// handle behaviour based on control char received
+	if (curState == "IDLE") {
+		if (ctrlChar == EOT)
+			LAST_EOT_RECEIVED = time(&LAST_EOT_RECEIVED);
```

Comment:
I think `LAST_EOT_RECEIVED` should just be set to `time(0)` to be assigned to the current time.